### ISSUE TITLE
Possible to run on unixy system?

### ISSUE THREAD

**Issue by xrisk** [loot/loot#610]
I run Skyrim through Wine. Can I use loot or is it restricted to Windows only?

**Comment by Freso** [comment]
LOOT should run fine under Wine… doesn't it? There has also been some work to make LOOT run under Linux (#491 and others), but actually making packages for it has not yet been successful. @WrinklyNinja is working on an automated build system that might also end up producing Linux builds (#607), so perhaps keep an eye on that.

**Comment by xrisk** [comment]
Alright, so I tried to run LOOT after the installer completed successfully, and:
```
Unhandled exception: unimplemented function KERNEL32.dll.WakeAllConditionVariable called in 32-bit code (0x7bc3f698).
Register dump:
 CS:001b SS:0023 DS:0023 ES:0023 FS:1007 GS:000f
 EIP:7bc3f698 ESP:0032f564 EBP:0032f5c8 EFLAGS:00000206(   - --  I   - -P- )
 EAX:006c0c6e EBX:0032f570 ECX:0032f5dc EDX:992735aa
 ESI:0032f600 EDI:006d3d2c
Stack dump:
0x0032f564:  0032f5e8 0060a4c0 0032f590 80000100
0x0032f574:  00000001 00000000 7bc3f698 00000002
0x0032f584:  006c0848 006c0c6e 7b8670d8 006d55fc
0x0032f594:  0032f5ac 7b818fa1 7b86b12b 00000090
0x0032f5a4:  00000000 0032f5b8 0032f5c8 0032f5d0
0x0032f5b4:  7b854800 0032f5f4 005ef0c9 0032f5dc
0200: sel=1007 base=7ffc0000 limit=00000fff 32-bit rw-
Backtrace:
=>0 0x7bc3f698 in <wine-loader> (+0x7bc3f698) (0x0032f5c8)
  1 0x0033000f in <wine-loader> (+0x33000e) (0x0032f5f4)
  2 0x005535c0 in loot (+0x1535bf) (0x0032f610)
  3 0x005536f4 in loot (+0x1536f3) (0x0032f654)
  4 0x0053deb1 in loot (+0x13deb0) (0x0032f690)
  5 0x0053dd08 in loot (+0x13dd07) (0x0032f6ac)
  6 0x0053dda5 in loot (+0x13dda4) (0x0032f6f8)
  7 0x004951fc in loot (+0x951fb) (0x0032fcbc)
  8 0x004af4c1 in loot (+0xaf4c0) (0x0032fce4)
  9 0x00533cdc in loot (+0x133cdb) (0x0032fd08)
  10 0x00518798 in loot (+0x118797) (0x0032fe64)
  11 0x005ef8c8 in loot (+0x1ef8c7) (0x0032feb0)
  12 0x7b8516ac in kernel32 (+0x416ab) (0x0032fec8)
  13 0x7b854867 in kernel32 (+0x44866) (0x0032ff08)
  14 0x7bc67efc in <wine-loader> (+0x7bc67efb) (0x0032ff28)
  15 0x7bc68faa in <wine-loader> (+0x7bc68fa9) (0x0032ffa8)
  16 0x7bc67ec2 in <wine-loader> (+0x7bc67ec1) (0x0032ffc8)
  17 0x7bc3fd3e in <wine-loader> (+0x7bc3fd3d) (0x0032ffe8)
0x7bc3f698: subl	$4,%esp
Modules:
Module	Address			Debug info	Name (37 modules)
ELF	       0-82001000	Stabs           <wine-loader>
PE	  400000-  6fc000	Deferred        loot
PE	10000000-1328b000	Deferred        libcef
PE	42510000-4254b000	Deferred        user32
PE	42660000-42667000	Deferred        gdi32
PE	44920000-44924000	Deferred        advapi32
PE	44990000-44994000	Deferred        version
PE	449b0000-44b0d000	Deferred        shell32
PE	44c00000-44c08000	Deferred        shlwapi
PE	44c80000-44caf000	Deferred        comctl32
PE	44da0000-44da8000	Deferred        ole32
PE	44ee0000-44ee4000	Deferred        rpcrt4
PE	44f70000-44f73000	Deferred        usp10
PE	44fb0000-44fb4000	Deferred        psapi
PE	44fc0000-44fc3000	Deferred        userenv
PE	44fe0000-44fe3000	Deferred        dhcpcsvc
PE	44ff0000-45001000	Deferred        urlmon
PE	450a0000-450a8000	Deferred        oleaut32
PE	451e0000-451f8000	Deferred        wininet
PE	45260000-4526a000	Deferred        mpr
PE	45280000-45287000	Deferred        winhttp
PE	452d0000-452d9000	Deferred        winspool
PE	45310000-453a8000	Deferred        comdlg32
PE	45420000-45430000	Deferred        oleacc
PE	45440000-454b7000	Deferred        winmm
PE	45500000-45509000	Deferred        msacm32
PE	45530000-45534000	Deferred        ws2_32
PE	45560000-45563000	Deferred        secur32
PE	45590000-45593000	Deferred        netapi32
PE	455c0000-455c4000	Deferred        iphlpapi
PE	455f0000-45630000	Deferred        crypt32
PE	456c0000-456c4000	Deferred        imm32
PE	456e0000-456e3000	Deferred        wtsapi32
PE	457d0000-457d3000	Deferred        winemac
PE	47800000-47804000	Deferred        uxtheme
PE	7b810000-7b9ad000	Deferred        kernel32
PE	7bc10000-7bc14000	Deferred        ntdll
Threads:
process  tid      prio (all id:s are in hex)
00000008 (D) Z:\Applications\Skyrim.app\Contents\Resources\drive_c\Program Files\LOOT\LOOT.exe
	00000009    0 <==
0000000e services.exe
	0000001e    0
	0000001d    0
	00000018    0
	00000016    0
	00000014    0
	00000010    0
	0000000f    0
00000012 winedevice.exe
	0000001c    0
	00000019    0
	00000017    0
	00000013    0
0000001a plugplay.exe
	00000020    0
	0000001f    0
	0000001b    0
00000021 explorer.exe
	00000023    0
	00000022    0
System information:
    Wine build: wine-1.6.1
    Platform: i386
    Host system: Darwin
    Host version: 15.5.0
```

**Comment by WrinklyNinja** [comment]
AFAIK LOOT won't run under Wine because its UI is Chromium-based, and Chromium doesn't run under Wine.

Native builds are possible, but nobody seems to have had much success replicating my test builds, and runtime dependencies have caused issues for people trying to run the builds I uploaded. I'm not interested in manually providing Linux builds, it's just not worth the effort to me.

#607 is not going to automate Linux builds though, only Windows builds. However, extending Travis to also do a full build and deploy is feasible: it's just not something I'm currently planning on doing.

What I am planning to do is update the Travis build process so that it builds and caches Boost like AppVeyor now does, as this would reduce the runtime dependencies (which would significantly speed up masterlist validation), and mean I could stop hacking around the ABI breakage. It would probably also have a side-effect of making Linux builds easier to use.

So in short, Linux binaries are still DIY, but I'm in favour of making it easier and/or automating it.

**Comment by Freso** [comment]
@xrisk Did you try running LOOT.exe with the `--no-sandbox` parameter as suggested at the very first sentende of #491?

**Comment by WrinklyNinja** [comment]
I've updated various bits and pieces to make building all targets (so including the LOOT application) possible on Travis, though my test run took 21 minutes vs. the usual 11. That's not too far off AppVeyor's ~17 minutes though, so I'll probably try setting up automated deployment of the binaries to Bintray at some point in the next week. Once that's done it'll be much easier to try them out and see if they work.

**Comment by WrinklyNinja** [comment]
There is now an automated Linux snapshot build [available](https://bintray.com/wrinklyninja/loot/metadata-validator#files) (Linux builds are the tar.xz files). I managed to run it on a freshly-installed Ubuntu 16.04 virtual machine, though I did have to launch it with `LD_LIBRARY_PATH=. ./LOOT` because it won't find the CEF shared library otherwise. I don't know how well it works otherwise, since I didn't have any of the supported games installed.

**Comment by Freso** [comment]
```
./LOOT: /usr/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by ./LOOT)
./LOOT: /usr/lib/libssl.so.1.0.0: no version information available (required by ./LOOT)
./LOOT: /usr/lib/libcrypto.so.1.0.0: no version information available (required by ./LOOT)
```

Guess I need to sort out some dependencies (or just figure out to get it building on Arch now that it's proven to be buildable at least). :)

**Comment by WrinklyNinja** [comment]
Automated Linux builds and deployments have been merged into the `dev` branch, and they run on a fresh install of Ubuntu 16.04 (I've fixed the `LD_LIBRARY_PATH` requirement), so I'm closing this as the answer to the issue's question is "yes"!

**Comment by Freso** [comment]
I still get
```
./LOOT: /usr/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by ./LOOT)
./LOOT: /usr/lib/libssl.so.1.0.0: no version information available (required by ./LOOT)
./LOOT: /usr/lib/libcrypto.so.1.0.0: no version information available (required by ./LOOT)
```
though. I guess I'll have to look into getting it compiling under Arch myself. :)

**Comment by WrinklyNinja** [comment]
Those are dependencies. The binaries are built on Ubuntu, I don't expect them to work on other flavours, because they won't have the same libraries installed by default or maybe even available.

**Comment by Freso** [comment]
Sure. Like I said, I'll have to look into compiling the binary under Arch itself, so it will be compiled with the system libraries:
```
freso@koume ~> ls -l /usr/lib/lib{curl,ssl,crypto}.so*
lrwxrwxrwx 1 root root      18  3 maj 18:12 /usr/lib/libcrypto.so -> libcrypto.so.1.0.0*
-r-xr-xr-x 1 root root 2851320  3 maj 18:12 /usr/lib/libcrypto.so.1.0.0*
lrwxrwxrwx 1 root root      16 21 jul 14:45 /usr/lib/libcurl.so -> libcurl.so.4.4.0*
lrwxrwxrwx 1 root root      23  1 jun 09:58 /usr/lib/libcurl.so.3 -> libcurl-compat.so.4.4.0*
lrwxrwxrwx 1 root root      16 21 jul 14:45 /usr/lib/libcurl.so.4 -> libcurl.so.4.4.0*
lrwxrwxrwx 1 root root      23  1 jun 09:58 /usr/lib/libcurl.so.4.0.0 -> libcurl-compat.so.4.4.0*
lrwxrwxrwx 1 root root      23  1 jun 09:58 /usr/lib/libcurl.so.4.1.0 -> libcurl-compat.so.4.4.0*
lrwxrwxrwx 1 root root      23  1 jun 09:58 /usr/lib/libcurl.so.4.2.0 -> libcurl-compat.so.4.4.0*
lrwxrwxrwx 1 root root      23  1 jun 09:58 /usr/lib/libcurl.so.4.3.0 -> libcurl-compat.so.4.4.0*
-rwxr-xr-x 1 root root  480792 21 jul 14:45 /usr/lib/libcurl.so.4.4.0*
lrwxrwxrwx 1 root root      15  3 maj 18:12 /usr/lib/libssl.so -> libssl.so.1.0.0*
-r-xr-xr-x 1 root root  512664  3 maj 18:12 /usr/lib/libssl.so.1.0.0*
```

**Comment by allista** [comment]
Answering to the original issue:
This version of LOOT for Windows does run under Wine:
https://www.dropbox.com/s/pthmbvea6xi9q1z/LOOT.v0.8.1.7z?dl=0
I don't remember where I've got it (it's been a year ago and I've searched a lot while bootstrapping my Skyrim installation). Maybe you could figure out why it's working; but even if you don't, people may at least use it as I do.

**Comment by wyrde** [comment]
LOOT Version 0.10.3 (build 0fcf788b) works in wine for me. It automatically picked up my skyrim install, drew the UI, and seems to be fully functional. Including running in mod organizer's environment and updating the plugins in the virtual filesystem.

The key was making sure loot can use libcef.dll and dwrite.dll (note, however, I have a bunch of libs and dlls installed for steam, skyrim, mod organizer, bash, bodyslide, and fnis).

In wine configuration for the container: dwrite is builtin,native and libcef is builtin,native (I had to add libcef).

Note this breaks steam's ability to draw its UI, but that's not important to me since I don't use steam for this container other than to auth skse and skyrim. That said, I'd definitely make sure you had everything else working before configuring wine for loot. (maybe I'll try steam w/o the -no-cef-sandbox flag in the future).

**Comment by C0rn3j** [comment]
Latest version I got to run was `LOOT 0.11.0` using `wine-staging 4.17`
Win version needs to be set to 7, which is the default.
No overrides are needed. 

Only issue I ran into is that I have to reinstall LOOT every time I want to run it, otherwise it crashes. Possibly fixable with some of the workarounds in the post above.

One thing that caught me off-guard was setting custom game path if using a different prefix - it's set in the empty column, not the last one - `Z:\home\c0rn3j\.local\share\Steam\steamapps\common\Skyrim Special Edition`
